### DISCUSSION TITLE
xmake检测系统库中是否存在某个变量

### DISCUSSION THREAD

**Discussion by jingkaimori** [xmake-io/xmake#3951] [Q&A]
目前有一个项目需要检查[`tzname`](https://linux.die.net/man/3/tzname)变量是否存在，Cmake中的`Check_Variable_Exists`函数和`Check_Symbol_Exists`可以检查某个变量是否被预定义，autotool生成的检查脚本如下：
```c
#include <time.h>
#if !HAVE_DECL_TZNAME
extern char *tzname[];
#endif

int
main ()
{
return tzname[0][0];
  ;
  return 0;
}
```
xmake有无这一功能？

**Comment by waruqi** [comment]
https://xmake.io/#/zh-cn/manual/helper_interfaces?id=detect-cc-code-snippets
